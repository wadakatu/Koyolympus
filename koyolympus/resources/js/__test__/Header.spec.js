import {shallowMount, RouterLinkStub} from '@vue/test-utils';
import Component from '@/HeaderComponent.vue';

describe('Testing methods', () => {
    test('Testing photo method', async () => {
        let mocks = {
            $store: {commit: jest.fn()},
            $router: {push: jest.fn().mockResolvedValue({})},
        }
        const wrapper = shallowMount(Component, {
            mocks,
            stubs: {
                RouterLink: RouterLinkStub
            },
        });
        await wrapper.vm.photo();
        expect(wrapper.vm.$store.commit).toHaveBeenCalledTimes(2);
        expect(wrapper.vm.$store.commit).toHaveBeenCalledWith('photo/setUrl', '/photo/random');
        expect(wrapper.vm.$store.commit).toHaveBeenCalledWith('photo/setGenre', null);
        expect(wrapper.vm.$router.push).toHaveBeenCalled();
        expect(wrapper.vm.$router.push).toHaveBeenCalledWith({name: 'photo.random'});
    });
});

describe('Testing @click event', () => {
    describe('Testing photo method', () => {
        test('call photo method when click `Photography`', () => {
            const mockPhoto = jest.spyOn(Component.methods, 'photo').mockImplementation(() => "mock photo");
            const wrapper = shallowMount(Component, {
                stubs: {
                    RouterLink: RouterLinkStub
                },
            });
            const target = wrapper.find('nav ul li.photo a');
            target.trigger('click');
            expect(target.text()).toBe('Photography');
            expect(mockPhoto).toHaveBeenCalledTimes(1);
        });
    });
});

describe('Testing router-link', () => {
    let wrapper;
    let links;
    beforeEach(() => {
        wrapper = shallowMount(Component, {
            stubs: {
                RouterLink: RouterLinkStub
            },
        });
        links = wrapper.findAllComponents(RouterLinkStub)
    });
    test('Testing home icon', () => {
        const target = links.at(0);
        expect(target.text()).toBe('');
        expect(target.props().to).toMatchObject({"name": "main"});
    });
    test('Testing About Me', () => {
        const target = links.at(1);
        expect(target.text()).toBe('About Me');
        expect(target.props().to).toMatchObject({"name": "about.me"});
    });
    test('Testing Photography', () => {
        const target = links.at(2);
        expect(target.text()).toBe('Photography');
        expect(target.props().to).toMatchObject({});
    });
    test('Testing Biz Inquiries', () => {
        const target = links.at(3);
        expect(target.text()).toBe('Biz Inquiries');
        expect(target.props().to).toMatchObject({"name": "main.biz"});
    });
});

describe('Testing snapshot', () => {
    test('default', () => {
        const wrapper = shallowMount(Component, {
            stubs: {
                RouterLink: RouterLinkStub
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
