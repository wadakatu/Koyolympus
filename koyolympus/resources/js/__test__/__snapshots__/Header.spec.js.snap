// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot default 1`] = `
<div
  class="navbar"
>
  <a>
    <img
      alt="myLogo"
      class="logo"
      src="/images/mylogo_white.png"
    />
  </a>
   
  <nav>
    <ul>
      <li
        class="about_me"
      >
        <a
          href=""
        >
          <a>
            About Me
          </a>
        </a>
      </li>
       
      <li
        class="photo"
      >
        <a
          href=""
        >
          <a>
            Photography
          </a>
        </a>
      </li>
       
      <li
        class="biz"
      >
        <a>
          <a>
            Biz Inquiries
          </a>
        </a>
      </li>
       
      <li
        class="more_menu"
      >
        <div
          class="dropdown"
        >
          <a
            class="dropbtn"
          >
            More
            <i
              class="fa fa-caret-down"
            />
          </a>
           
          <div
            class="dropdown-content"
          >
            <a
              href="https://koyolympus.thebase.in/"
              rel="noopener noreferrer"
              target="_blank"
            >
              E-Commerce
            </a>
             
            <a
              href="https://www.facebook.com/koyolympus/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Facebook
            </a>
             
            <a
              href="https://github.com/wadakatu/koyolympus"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </a>
             
            <a
              href="https://www.instagram.com/wadakatu1234/?hl=ja"
              rel="noopener noreferrer"
              target="_blank"
            >
              Instagram
            </a>
          </div>
        </div>
      </li>
    </ul>
  </nav>
   
  <div
    class="hamburger"
    id="menuArea"
  >
    <input
      id="menuToggle"
      type="checkbox"
    />
     
    <label
      class="menuOpen"
      for="menuToggle"
    >
      <span
        class="open"
      />
    </label>
     
    <div
      class="menu menuEffects"
    >
      <label
        for="menuToggle"
      />
       
      <div
        class="menuContent"
      >
        <ul>
          <li>
            <a>
              About Me
            </a>
          </li>
           
          <li>
            <a>
              Photography
            </a>
          </li>
           
          <li>
            <a>
              Biz Inquiries
            </a>
          </li>
           
          <li>
            <a
              href="https://koyolympus.thebase.in/"
              rel="noopener noreferrer"
              target="_blank"
            >
              E-Commerce
                    
            </a>
          </li>
           
          <li>
            <a
              href="https://www.facebook.com/koyolympus/"
              rel="noopener noreferrer"
              target="_blank"
            >
              
                        Facebook
                    
            </a>
          </li>
           
          <li>
            <a
              href="https://github.com/wadakatu/koyolympus"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
                    
            </a>
          </li>
           
          <li>
            <a
              href="https://www.instagram.com/wadakatu1234/?hl=ja"
              rel="noopener noreferrer"
              target="_blank"
            >
              Instagram
                    
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
